src/main/java/seedu/address/storage/JsonAdaptedInternship.java

```diff
@@ -0,0 +1,128 @@
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted internship object into the model's {@code Internship} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted internship.
+     */
+    public Internship toModelType() throws IllegalValueException {
+        if (internshipId == null) {
+            throw new IllegalValueException(
+                    String.format(MISSING_FIELD_MESSAGE_FORMAT, InternshipId.class.getSimpleName()));
+        }
+        if (!InternshipId.isValidId(internshipId)) {
+            throw new IllegalValueException(InternshipId.MESSAGE_CONSTRAINTS);
+        }
+        final InternshipId modelInternshipId = new InternshipId(internshipId);
+
+        if (companyName == null) {
+            throw new IllegalValueException(
+                    String.format(MISSING_FIELD_MESSAGE_FORMAT, CompanyName.class.getSimpleName()));
+        }
+        if (!CompanyName.isValidName(companyName)) {
+            throw new IllegalValueException(CompanyName.MESSAGE_CONSTRAINTS);
+        }
+        final CompanyName modelCompanyName = new CompanyName(companyName);
+
+        if (internshipRole == null) {
+            throw new IllegalValueException(
+                    String.format(MISSING_FIELD_MESSAGE_FORMAT, InternshipRole.class.getSimpleName()));
+        }
+        if (InternshipRole.isValidName(internshipRole)) {
+            final InternshipRole modelInternshipRole = new InternshipRole(internshipRole);
+
+            if (internshipStatus == null) {
+                throw new IllegalValueException(
+                        String.format(MISSING_FIELD_MESSAGE_FORMAT, InternshipStatus.class.getSimpleName()));
+            }
+            if (!InternshipStatus.isValidStatus(internshipStatus)) {
+                throw new IllegalValueException(InternshipStatus.MESSAGE_CONSTRAINTS);
+            }
+            final InternshipStatus modelInternshipStatus =
+                    new InternshipStatus(InternshipStatus.State.valueOf(internshipStatus));
+
+            final PersonId modelContactPersonId;
+            if (contactPersonId == null || !InternshipId.isValidId(contactPersonId)) {
+                modelContactPersonId = null;
+            } else {
+                modelContactPersonId = new PersonId(contactPersonId);
+            }
+
+            InterviewDate modelInterviewDate;
+            if (interviewDate == null || !InterviewDate.isValidDatetimeStr(interviewDate)) {
+                modelInterviewDate = null;
+            } else {
+                modelInterviewDate = new InterviewDate(interviewDate);
+            }
+
+            return new Internship(
+                    modelInternshipId,
+                    modelCompanyName,
+                    modelInternshipRole,
+                    modelInternshipStatus,
+                    modelContactPersonId,
+                    modelInterviewDate);
```

Comment:
Is there a sudden change in the handling method? Is there a need for InternshipStatus checks to be within InternshipRole check

Reply:
Ohh yup that's not intended, they should be separate. Thanks for pointing out!